static/app/components/searchQueryBuilder/index.spec.tsx

```diff
@@ -221,6 +221,33 @@ describe('SearchQueryBuilder', function () {
         screen.queryByRole('button', {name: 'Clear search query'})
       ).not.toBeInTheDocument();
     });
+
+    it('is hidden if the prop is specified and text is empty', async function () {
+      const mockOnChange = jest.fn();
+      render(
+        <SearchQueryBuilder
+          {...defaultProps}
+          onChange={mockOnChange}
+          hideClearWhenEmpty
+        />
+      );
+      // Must await something to prevent act warnings
+      await act(tick);
```

Comment:
just put a await findBy before other things
```suggestion
      await screen.findByRole('combobox', {name: 'Add a search term'});
```